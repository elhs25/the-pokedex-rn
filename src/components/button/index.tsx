import FAButton from "./FAButton/FAButton";

export {
    FAButton
}