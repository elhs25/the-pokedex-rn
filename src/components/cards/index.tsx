import PokemonCardPicture from "./PokemonCardPicture/PokemonCardPicture";

export {
    PokemonCardPicture
}