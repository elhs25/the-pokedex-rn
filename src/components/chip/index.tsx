import PokemonTypeChip from "./PokemonTypeChip/PokemonTypeChip";

export {
    PokemonTypeChip
}