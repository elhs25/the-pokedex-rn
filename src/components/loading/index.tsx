import FooterLoadingIndicator from "./FooterLoadingIndicator/FooterLoadingIndicator";

export {
    FooterLoadingIndicator
}