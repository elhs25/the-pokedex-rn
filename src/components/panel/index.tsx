import StatsChart from "./StatsChart/StatsChart";

export {
    StatsChart
}