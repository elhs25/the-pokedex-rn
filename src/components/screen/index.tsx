import ScreenContainer, { ScreenContent } from "./ScreenContainer/ScreenContainer";

export {
    ScreenContainer,
    ScreenContent,
}