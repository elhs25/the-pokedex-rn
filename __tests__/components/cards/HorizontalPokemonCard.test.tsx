import React from "react";

import {it, expect, describe} from '@jest/globals';
import renderer from 'react-test-renderer';
import { HorizontalPokemonCard } from "components/cards";

describe('HorizontalPokemonCard Component', () => {

    it('renders correctly', () => {
        const Component = renderer.create(
            <HorizontalPokemonCard  
                name="mudkip"
                id={258}
                sprite="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/258.png"
                onPress={() => {}}
            />
        ).toJSON();
        expect(Component).toMatchSnapshot();
    });
});