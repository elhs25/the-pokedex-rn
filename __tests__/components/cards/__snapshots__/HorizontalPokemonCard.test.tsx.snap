// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalPokemonCard Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "elevation": 5,
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
      "shadowColor": "#616161",
      "shadowOffset": {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "borderRadius": 8,
        "height": 70,
        "marginRight": 12,
        "padding": 4,
        "width": 70,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/258.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "auto",
        }
      }
    />
  </View>
  <View>
    <Text
      style={
        [
          {
            "color": "#616161",
            "fontSize": 24,
            "textTransform": "capitalize",
          },
          {
            "fontWeight": "bold",
            "marginBottom": 8,
          },
        ]
      }
    >
      mudkip
    </Text>
    <Text
      style={
        {
          "color": "#616161",
          "fontSize": 24,
          "textTransform": "capitalize",
        }
      }
    >
      #0258
    </Text>
  </View>
</View>
`;
